src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -132,21 +133,40 @@ IWeakReference* GetWinRTWeakReference(OBJECTREF* pObject)
 
     MethodTable* pMT = (*pObject)->GetMethodTable();
 
-    // If the object is not an RCW, then we do not want to use a WinRT weak reference to it
-    if (!pMT->IsComObjectType())
-    {
-        return nullptr;
-    }
+    IWeakReferenceSource* pWeakReferenceSourceUnsafe;
```

Comment:
Naked COM interfaces should almost never happen nowadays. This should be: `SafeComHolder<IWeakReferenceSource> weakReferenceSource;`